File: backend/src/models/CartItem.js
const mongoose = require('mongoose');

const { Schema } = mongoose;

const cartItemSchema = new Schema(
  {
    cart: {
      type: Schema.Types.ObjectId,
      ref: 'Cart',
      required: true,
      index: true,
    },
    product: {
      type: Schema.Types.ObjectId,
      ref: 'Product',
      required: true,
      index: true,
    },
    quantity: { type: Number, required: true, min: 1, default: 1 },
    priceAtAddition: { type: Number, required: true, min: 0 },
  },
  { timestamps: true }
);

cartItemSchema.index({ cart: 1, product: 1 }, { unique: true });

module.exports = mongoose.model('CartItem', cartItemSchema);

File: backend/src/models/CartItem.py
from __future__ import annotations

from beanie import Indexed
from beanie.odm.fields import PydanticObjectId
from pydantic import Field
from pymongo import IndexModel

from .base import TimeStampedDocument


class CartItem(TimeStampedDocument):
    cart: Indexed(PydanticObjectId)  # type: ignore[assignment]
    product: Indexed(PydanticObjectId)  # type: ignore[assignment]
    quantity: int = Field(default=1, ge=1)
    priceAtAddition: float = Field(ge=0)

    class Settings:
        name = "cart_items"
        use_revision = False
        indexes = [IndexModel([("cart", 1), ("product", 1)], unique=True)]
